keps/sig-testing/2420-reducing-kubernetes-build-maintenance/README.md

```diff
@@ -0,0 +1,687 @@
+#### Alpha -> Beta Graduation
+
+- Gather feedback from developers and surveys
+- Complete features A, B, C
+- Tests are in Testgrid and linked in KEP
+
+#### Beta -> GA Graduation
+
+- N examples of real-world usage
+- N installs
+- More rigorous forms of testing—e.g., downgrade tests and scalability tests
+- Allowing time for feedback
+
+**Note:** Generally we also wait at least two releases between beta and
+GA/stable, because there's no opportunity for user feedback, or even bug reports,
+in back-to-back releases.
+
+#### Removing a Deprecated Flag
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality that deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+
+**For non-optional features moving to GA, the graduation criteria must include 
+[conformance tests].**
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+-->
+
+This will be declared stable/GA when:
+- All kubernetes/kubernetes CI jobs use the preexisting make build system
+- Bazel-related source files and related tooling are removed from the kubernetes/kubernetes repository on currently supported release branches and the current development branch
+- Bazel-related configuration/presets are removed from kubernetes/kubernetes CI jobs in kubernetes/test-infra
+
+
+As bazel-built artifacts are not built or distributed as part of a kubernetes/kubernetes release, there is no deprecation window required.
+
+### Upgrade / Downgrade Strategy
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to maintain previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to make use of the enhancement?
+-->
+
+n/a. Not relevant to upgrades. Existing release builds and upgrade CI use make.
+
+### Version Skew Strategy
+
+<!--
+If applicable, how will the component handle version skew with other
+components? What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- Does this enhancement involve coordinating behavior in the control plane and
+  in the kubelet? How does an n-2 kubelet without this feature available behave
+  when this feature is used?
+- Will any other components on the node change? For example, changes to CSI,
+  CRI or CNI may require updating that component before the kubelet.
+-->
+
+n/a.
+
+## Production Readiness Review Questionnaire
+
+**TODO**: This entire section seems completely irrelevant for KEPs that do not
```

Comment:
this is TBD, I'm not sure what the policy is here, but I don't think PRR is actually relevant to a KEP like this with literally no change to release artifacts.

Reply:
Discussing KEP types here: https://github.com/kubernetes/enhancements/issues/2311
Drop some notes when you have a chance! :)

Reply:
I'll take another pass at these fields and the metadata in general tomorrow, and definitely take a look at #2311

Reply:
I suggested edits to drop the text under each heading to "n/a" with explanation as to why, but we can address as followup